src/waveform/renderers/glslwaveformrenderersignal.cpp

```diff
@@ -238,19 +237,23 @@ void GLSLWaveformRendererSignal::onSetTrack() {
 }
 
 void GLSLWaveformRendererSignal::onResize() {
+    // onInitializeGL not called yet
+    if (!m_frameShaderProgram) {
+        return;
+    }
     createFrameBuffers();
 }
 
 void GLSLWaveformRendererSignal::slotWaveformUpdated() {
     m_textureRenderedWaveformCompletion = 0;
+    // onInitializeGL not called yet
+    if (!m_frameShaderProgram) {
+        return;
+    }
     loadTexture();
 }
 
 void GLSLWaveformRendererSignal::draw(QPainter* painter, QPaintEvent* /*event*/) {
-    if (!m_framebuffer || !m_framebuffer->isValid() || !m_shadersValid) {
-        return;
-    }
-
```

Comment:
Why was this removed? Is it really necessary?

Reply:
This would prevent reaching the `maybeInitializeGL` call below.

Reply:
OK.